### PR TITLE
Merge `master` into `dev/feature`

### PR DESCRIPTION
**Pull request by APickledWalrus**
### Description
This PR merges `master` into `dev/feature`. Conflicts have been resolved and a few additional corrections made. I have reviewed all changed files but another set of eyes would be good.

---
**Target Minecraft Versions:** any 
**Requirements:** none 
**Related Issues:** none

### CODE REVIEW

File: src/main/java/ch/njol/skript/structures/StructFunction.java [hunk 1]
```diff
@@ -58,45 +60,56 @@ public class StructFunction extends Structure {
...
 	public static final Priority PRIORITY = new Priority(400);
 
+	private static final Pattern SIGNATURE_PATTERN =
+			Pattern.compile("(?:local )?function (" + Functions.functionNamePattern + ")\\((.*)\\)(?:\\s*::\\s*(.+))?");
 	private static final AtomicBoolean VALIDATE_FUNCTIONS = new AtomicBoolean();
 
 	static {
 		Skript.registerStructure(StructFunction.class,
-			"[:local] function <(" + Functions.functionNamePattern + ")\\((.*)\\)(?:\\s*(?:::| returns )\\s*(.+))?>"
+			"[:local] function <.+>"
```
Comment (Fusezion) [inline src/main/java/ch/njol/skript/structures/StructFunction.java:69]: So uh rip `blah() returns blah`, forgot to add `returns` into the new static field

**Review by Fusezion — CHANGES_REQUESTED** [review]
didn't see much but I noticed this when I was doing a self build of the exact same thing

File: src/main/java/ch/njol/skript/structures/StructFunction.java (hunk 1 above)
Comment (APickledWalrus) [reply src/main/java/ch/njol/skript/structures/StructFunction.java:69]: nice catch!

**Comment by sovdeeth** [comment]
Test failures seem to be due to a change with drops in 1.20.2, see https://discord.com/channels/135877399391764480/836220422223036467/1181789049124569161

**Review by sovdeeth — APPROVED** [review]

**Review by Moderocky — APPROVED** [review]

**Pull request merged**